easybuild/easyblocks/g/gromacs.py

```diff
@@ -498,3 +554,115 @@ def sanity_check_step(self):
+                'double': '-DGMX_DOUBLE=ON',
+            }
+            mpi_type_opts = {
+                'nompi': '-DGMX_MPI=OFF -DGMX_THREAD_MPI=ON',
+                'mpi': '-DGMX_MPI=ON -DGMX_THREAD_MPI=OFF'
+            }
+
+        if LooseVersion(self.version) < LooseVersion('5'):
+            # For older versions we only build/install the mdrun part for
+            # the MPI variant. So we need to be able to specify the
+            # install target depending on variant.
+            self.cfg['install_cmd'] = 'make'
+
+            # Use the fact that for older versions we just need to
+            # build and install mdrun for the MPI part
+            build_opts = {
+                'nompi': '',
+                'mpi': 'mdrun'
+            }
+            install_opts = {
+                'nompi': 'install',
+                'mpi': 'install-mdrun'
+            }
+        else:
+            build_opts = {
+                'nompi': '',
+                'mpi': ''
+            }
+            install_opts = {
+                'nompi': 'install',
+                'mpi': 'install'
+            }
+
+        precs = ['single']
+        if self.cfg['double_precision'] is None or self.cfg['double_precision']:
+            precs.append('double')
+
+        mpitypes = ['nompi']
+        if self.toolchain.options.get('usempi', None):
+            mpitypes.append('mpi')
+
+        # We need to count the numb er of variations to build.
+        self.variants_to_build = 0
+        versions_built = []
+        # Handle the different variants
+        for prec in precs:
+            for mpitype in mpitypes:
+                self.variants_to_build += 1
+                versions_built.append('%s precision %s' % (prec, mpitype))
+                var_confopts = []
+                var_buildopts = []
+                var_installopts = []
+
+                var_confopts.append(mpi_type_opts[mpitype])
+                var_confopts.append(prec_opts[prec])
+                if LooseVersion(self.version) < LooseVersion('4.6'):
+                    suffix = ''
+                    if mpitype == 'mpi':
+                        suffix = "--program-suffix={0}".format(self.cfg['mpisuffix'])
+                        if prec == 'double':
+                            suffix += '_d'
+                    var_confopts.append(suffix)
+
+                var_buildopts.append(build_opts[mpitype])
+                var_installopts.append(install_opts[mpitype])
+
+                self.cfg.update('configopts', [' '.join(var_confopts) + ' ' + common_config_opts])
+                self.cfg.update('buildopts', [' '.join(var_buildopts) + ' ' + common_build_opts])
+                self.cfg.update('installopts', [' '.join(var_installopts) + ' ' + common_install_opts])
+
+        self.log.debug("List of configure options to iterate over: %s", self.cfg['configopts'])
+        print_msg("Building these versions of GROMACS: %s" % ', '.join(versions_built), silent=self.silent)
```

Comment:
Isn't a verbose (stdout shown) message unusal? Why not `self.log.info`?

Reply:
To make the user aware that it is going to run x number of builds.

Reply:
Doesn't EB does this already? I remember "Running iteration n" messages

Reply:
Yes it does but it doesn't tell you why

Reply:
So does none of the other ECs/EBs. You have to look into the log for that.
I'm just mentioning this out of consistency. There has to be a guideline on what to print where IMO or the stdout will be cluttered to unreadability.